lib/Mojolicious/Plugin/CGI.pm

```diff
@@ -215,10 +215,14 @@ sub register {
         $| = 1;
         select STDOUT;
         $| = 1;
-        { exec $self->{script} }
-        die "Could not execute $self->{script}: $!";
+	if ($self->{run}) {
+	  $self->{run}->();
```

Comment:
How about this:

    If (my $code = $self->{run}) {
      $c->$code($self);
      exit;
    }